cmd/oras/internal/option/remote.go

```diff
@@ -305,9 +309,11 @@ func (opts *Remote) NewRepository(reference string, common Common, logger logrus
 
 // isPlainHttp returns the plain http flag for a given registry.
 func (opts *Remote) isPlainHttp(registry string) bool {
-	host, _, _ := net.SplitHostPort(registry)
-	if host == "localhost" || registry == "localhost" {
-		return true
+	if plainHTTP, enforced := opts.plainHTTP(); enforced {
+		return plainHTTP
 	}
-	return opts.PlainHTTP
+
+	// not specified, defaults to plain http for localhost
+	host, _, _ := net.SplitHostPort(registry)
+	return host == "localhost" || registry == "localhost"
```

Comment:
This part seems not right.

Should it be
```suggestion
	// not specified, defaults to plain http for localhost
	host, _, _ := net.SplitHostPort(registry)
	if host == "localhost" || registry == "localhost" {
		return true
	}
	return plainHTTP
```
?